lib/SimpleSAML/Kernel.php

```diff
@@ -1,170 +1,69 @@
-    {
-        return [
-            new FrameworkBundle(),
-        ];
-    }
-
-
-    /**
-     * Get the module loaded in this kernel.
-     *
-     * @return string
-     */
-    public function getModule(): string
-    {
-        return $this->module;
-    }
-
-
-    /**
-     * Configures the container.
-     *
-     * @param ContainerBuilder $container
-     * @param LoaderInterface $loader
-     * @return void
-     */
-    protected function configureContainer(ContainerBuilder $container, LoaderInterface $loader): void
+    protected function configureRoutes(RouteCollectionBuilder $routes)
     {
-        $configuration = Configuration::getInstance();
-        $baseDir = $configuration->getBaseDir();
-        $loader->load($baseDir . '/routing/services/*' . self::CONFIG_EXTS, 'glob');
-        $confDir = Module::getModuleDir($this->module) . '/routing/services';
-        if (is_dir($confDir)) {
-            $loader->load($confDir . '/**/*' . self::CONFIG_EXTS, 'glob');
+        foreach ($this->bundles as $bundle) {
+            if ($bundle instanceof \SimpleSAML\Modules\Module) {
+                $baseDir = $bundle->getPath() . '/Resources/config';
+                $name = strtolower($bundle->getName());
+                $prefix = substr(
+                    $name,
+                    0,
+                    strpos($name, 'module'));
+
+                if (is_dir($baseDir)) {
+                    $routes->import($baseDir . '/routes' . self::CONFIG_EXTS, $prefix, 'glob');
+                }
+            }
         }
 
-        $container->loadFromExtension('framework', [
-            'secret' => Configuration::getInstance()->getString('secretsalt'),
-        ]);
-
-        $this->registerModuleControllers($container);
-    }
+        $confDir = __DIR__ . '/Resources/config';
 
+        $routes->import($confDir.'/{routes}/'.$this->environment.'/**/*'.self::CONFIG_EXTS, '/', 'glob');
+        $routes->import($confDir.'/{routes}/*'.self::CONFIG_EXTS, '/', 'glob');
+        $routes->import($confDir.'/{routes}'.self::CONFIG_EXTS, '/', 'glob');
 
-    /**
-     * Import routes.
-     *
-     * @param RouteCollectionBuilder $routes
-     * @return void
-     */
-    protected function configureRoutes(RouteCollectionBuilder $routes): void
-    {
-        $configuration = Configuration::getInstance();
-        $baseDir = $configuration->getBaseDir();
-        $routes->import($baseDir . '/routing/routes/*' . self::CONFIG_EXTS, '/', 'glob');
-        $confDir = Module::getModuleDir($this->module) . '/routing/routes';
-        if (is_dir($confDir)) {
-            $routes->import($confDir . '/**/*' . self::CONFIG_EXTS, $this->module, 'glob');
-        }
     }
 
-
-    /**
-     * @param ContainerBuilder $container
-     * @return void
-     */
-    private function registerModuleControllers(ContainerBuilder $container): void
+    protected function configureContainer(ContainerBuilder $container, LoaderInterface $loader)
```

Comment:
Load services required by the core of simplesaml (modules services are loaded inside the Module loader class).